libraries/rush-lib/scripts/copyEmptyModules.js

```diff
@@ -46,39 +46,44 @@ module.exports = {
-      const folderItems = await FileSystem.readFolderItemsAsync(
-        relativeFolderPath ? `${jsInFolderPath}/${relativeFolderPath}` : jsInFolderPath
-      );
-      await Async.forEachAsync(folderItems, async (folderItem) => {
-        const itemName = folderItem.name;
-        const relativeItemPath = relativeFolderPath ? `${relativeFolderPath}/${itemName}` : itemName;
+    const folderPathQueue = new AsyncQueue([undefined]);
 
-        if (folderItem.isDirectory()) {
-          await searchAsync(relativeItemPath);
-        } else if (folderItem.isFile() && itemName.endsWith(JS_FILE_EXTENSION)) {
-          const jsInPath = `${jsInFolderPath}/${relativeItemPath}`;
-          const jsFileText = await FileSystem.readFileAsync(jsInPath);
-          const strippedJsFileText = stripCommentsFromJsFile(jsFileText);
-          if (strippedJsFileText === 'export {};') {
-            const outJsPath = `${outFolderPath}/${relativeItemPath}`;
-            terminal.writeVerboseLine(`Writing stub to ${outJsPath}`);
-            await FileSystem.writeFileAsync(outJsPath, 'module.exports = {};', {
-              ensureFolderExists: true
-            });
+    await Async.forEachAsync(
+      folderPathQueue,
+      async ([relativeFolderPath, callback]) => {
+        const folderPath = relativeFolderPath ? `${jsInFolderPath}/${relativeFolderPath}` : jsInFolderPath;
+        const folderItems = await FileSystem.readFolderItemsAsync(folderPath);
+        for (const folderItem of folderItems) {
+          const itemName = folderItem.name;
+          const relativeItemPath = relativeFolderPath ? `${relativeFolderPath}/${itemName}` : itemName;
 
-            const relativeDtsPath = relativeItemPath.slice(0, -JS_FILE_EXTENSION.length) + DTS_FILE_EXTENSION;
-            const inDtsPath = `${dtsInFolderPath}/${relativeDtsPath}`;
-            const outDtsPath = `${outFolderPath}/${relativeDtsPath}`;
-            terminal.writeVerboseLine(`Copying ${inDtsPath} to ${outDtsPath}`);
-            // We know this is a file, don't need the redundant checks in FileSystem.copyFileAsync
-            const buffer = await FileSystem.readFileToBufferAsync(inDtsPath);
-            await FileSystem.writeFileAsync(outDtsPath, buffer, { ensureFolderExists: true });
+          if (folderItem.isDirectory()) {
+            folderPathQueue.push(relativeItemPath);
+          } else if (folderItem.isFile() && itemName.endsWith(JS_FILE_EXTENSION)) {
+            const jsInPath = `${jsInFolderPath}/${relativeItemPath}`;
+            const jsFileText = await FileSystem.readFileAsync(jsInPath);
+            const strippedJsFileText = stripCommentsFromJsFile(jsFileText);
+            if (strippedJsFileText === 'export {};') {
+              const outJsPath = `${outFolderPath}/${relativeItemPath}`;
+              terminal.writeVerboseLine(`Writing stub to ${outJsPath}`);
+              await FileSystem.writeFileAsync(outJsPath, 'module.exports = {};', {
```

Comment:
Minor nit, for performance we should store the result of `Buffer.from('module.exports = {};', 'utf8')` and reuse teh same buffer for all of these calls.